processing/src/main/java/org/apache/druid/query/filter/BooleanFilter.java

```diff
@@ -23,13 +23,14 @@
 import org.apache.druid.segment.ColumnSelectorFactory;
 
 import java.util.HashSet;
+import java.util.List;
 import java.util.Set;
 
 public interface BooleanFilter extends Filter
 {
   ValueMatcher[] EMPTY_VALUE_MATCHER_ARRAY = new ValueMatcher[0];
 
-  Set<Filter> getFilters();
+  List<Filter> getFilters();
```

Comment:
Could you add a javadoc to this function explaining why we a list is used here so that we can preserve the order of the filters to enable short-circuiting.

Writing out this comment, made me think perhaps we could achieve the same outcome by using a LinkedHashSet instead - that would make it clearer in the rest of the code that there are no duplicates in the filters

Reply:
I think doing the LinkedHashSet makes sense, so I went with that approach.